### ISSUE TITLE
Feature: Preprocessing

### ISSUE THREAD

**Issue by DiogenesAnalytics** [DiogenesAnalytics/autoencoder#18]
# Problem
Currently there is no builtin method for handling the *preprocessing* of `image`/`text` data.

## References
+ https://keras.io/2.15/api/layers/preprocessing_layers/
+ https://www.tensorflow.org/guide/keras/preprocessing_layers
+ https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing
+ https://keras.io/2.15/api/data_loading/
+ https://keras.io/2.15/api/utils/feature_space/
+ https://www.youtube.com/watch?v=GVShIIh3_yE
+ https://github.com/keras-team/keras-preprocessing
+ https://keras.io/api/datasets/mnist/#load_data-function
+ https://blog.keras.io/building-autoencoders-in-keras.html
+ https://medium.com/analytics-vidhya/a-tip-a-day-python-tip-8-why-should-we-normalize-image-pixel-values-or-divide-by-255-4608ac5cd26a
+ https://stats.stackexchange.com/questions/211436/why-normalize-images-by-subtracting-datasets-image-mean-instead-of-the-current
+ https://stackoverflow.com/questions/62111708/what-is-the-significance-of-normalization-of-data-before-feeding-it-to-a-ml-dl-m
+ https://stats.stackexchange.com/questions/185853/why-do-we-need-to-normalize-the-images-before-we-put-them-into-cnn
+ https://becominghuman.ai/image-data-pre-processing-for-neural-networks-498289068258
+ https://forums.fast.ai/t/images-normalization/4058
+ https://keras.io/2.15/api/utils/python_utils/#normalize-function
+ https://kozodoi.me/blog/20210308/compute-image-stats
+ https://stackoverflow.com/a/60592637
+ https://keras.io/api/layers/reshaping_layers/reshape/
